pkg/controller/vitessshard/reconcile_tablets.go

```diff
@@ -261,6 +261,9 @@ func vttabletSpecs(vts *planetscalev2.VitessShard, parentLabels map[string]strin
 			labels[planetscalev2.TabletTypeLabel] = string(pool.Type)
 			labels[planetscalev2.TabletIndexLabel] = strconv.FormatUint(uint64(tabletIndex), 10)
 
+			// Merge ExtraVitessFlags into the tablet spec ExtraFlags field.
+			update.StringMap(&pool.Vttablet.ExtraFlags, vts.Spec.ExtraVitessFlags)
```

Comment:
We should make a new map like the others, to avoid mutating inputs.

Reply:
Isn't the goal in this specific case to mutate ExtraFlags on VttabletSpec?

Reply:
Done https://github.com/planetscale/vitess-operator/pull/14/commits/74fb0d0aa4f38f826d38612baf778c94c0ed6095.